### ISSUE TITLE
[Rule Request] Count Where

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2617]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Prefer using `count(where:)` over `filter { }.count`.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

https://github.com/apple/swift-evolution/blob/master/proposals/0220-count-where.md

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// bad
[1, 2, 3, -1, -2].filter({ $0 > 0 }).count
[1, 2, 3, -1, -2].filter { $0 > 0 }.count
[1, 2, 3, -1, -2].lazy.filter({ $0 > 0 }).count

// good 
[1, 2, 3, -1, -2].count(where: { $0 > 0 })
[1, 2, 3, -1, -2].count { $0 > 0 }
```
3. Should the rule be configurable, if so what parameters should be configurable?

Just severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in as `count(where:)` is only available on Swift 5+

**Comment by rounak** [comment]
I can work on this

**Comment by marcelofabri** [comment]
https://github.com/apple/swift/pull/22289 😭

**Comment by marcelofabri** [comment]
Yeah, I don’t think it makes sense to add the rule for now
